### PR TITLE
Fix ordering of manifest diff output

### LINKED ISSUE #16834: manifest diff ordering should be stable
When multiple resources are different, the ordering changes non-deterministically. 
Ideally, the ordering should be grouped by kind, then namespace and name - this can be easily achieved by sorting the resource keys alphabetically.

### PR DESCRIPTION
**Pull request by ostromart**
Fixes https://github.com/istio/istio/issues/16834.

### CODE REVIEW

File: pkg/object/objects.go [hunk 1]
```diff
@@ -465,17 +466,27 @@ func manifestDiff(aom, bom map[string]*K8sObject, im map[string]string) (string,
...
-			writeStringSafe(&sb, "\n\nObject "+ak+" has diffs:\n\n")
-			writeStringSafe(&sb, diff)
+			out[ak] = fmt.Sprint("\n\nObject "+ak+" has diffs:\n\n", diff)
 		}
 	}
 	for bk := range bom {
 		ao := aom[bk]
 		if ao == nil {
-			writeStringSafe(&sb, "\n\nObject "+bk+" is missing in A:\n\n")
+			out[bk] = fmt.Sprint("\n\nObject " + bk + " is missing in B:\n\n")
```
Comment (morvencao) [inline pkg/object/objects.go]: `B` -> `A` ?

Comment (ostromart) [reply pkg/object/objects.go]: Done.

**Review by richardwxn — APPROVED** [review]

**Review by elfinhe — APPROVED** [review]
LGTM

File: pkg/object/objects.go [hunk 2]
```diff
@@ -448,14 +448,15 @@ func buildResourceRegexp(s string) (*regexp.Regexp, error) {
...
+	out := make(map[string]string)
 	for ak, av := range aom {
 		ay, err := av.YAML()
 		if err != nil {
 			return "", err
 		}
 		bo := bom[ak]
 		if bo == nil {
-			writeStringSafe(&sb, "\n\nObject "+ak+" is missing in B:\n\n")
+			out[ak] = fmt.Sprint("\n\nObject " + ak + " is missing in B:\n\n")
```
Comment (ayj) [inline pkg/object/objects.go]: `fmt.Sprint` or string concatenation but not both at the same time, e.g.

```golang
out[ak] = `\n\nObject ` + ak + ` is missing in B:\n\n`
// or
out[ak] = fmt.Sprintf("`\n\nObject %v is missing in B:\n\n`, ak)
```

Comment (ostromart) [reply pkg/object/objects.go]: Done, thanks.

**Review by ayj — APPROVED** [review]

**Pull request merged**